### PR TITLE
#CP-39: User update own profile

### PR DESCRIPTION
**Pull request by feyton**
### What does this PR do?

Allow a user to edit their own profile from the frontend using their access_token to send that

### Description of Task to be completed

- [x] User can upload profile picture
- [x] User can change their number 

### How should this be manually tested?

```
git clone https://github.com/atlp-rwanda/phantom-be-codebandits
cd phantom-be-codebandits
git fetch origin
git checkout <your branch>
npm install
//configure the .env according to the .env.sample file
// start the database
npm run dev
// open browser or postman (Check linked public workplace for testing)
```

### Any background context you want to provide?

The picture uploads are handled with cloudinary. A cloudinary account is required to be able to upload the picture

### What are the relevant pivotal tracker stories?

[#CP-39

### CODE REVIEW

**Review by Abdulkeza — DISMISSED** [review]
The best feature of using Cloudinary is working fine

File: src/busesApp/controllers.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const getBusHandler = async (req, res) => {
 	let options;
 	// eslint-disable-next-line no-unused-expressions
 	req.query?.relation === 'true'
-		? (options = { relations: ['route'] })
+		? (options = { relations: ['route', 'driver'] })
```
Comment (Yvad60) [inline src/busesApp/controllers.js:28]: Good job with this addition, these relations are crucial

File: src/operatorApp/tests/operator.test.js [hunk 2]
```diff
@@ -48,6 +48,14 @@ describe('Operator router tests', () => {
...
 		expect(response).to.have.status(200);
 	});
+	it('should update admin profile', async () => {
+		const response = await chai
+			.request(app)
+			.post('/api/v1/accounts/profile')
+			.send({ firstName: 'Feyton' })
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(200);
+	});
```
Comment (Yvad60) [inline src/operatorApp/tests/operator.test.js:58]: did you confirm that the profile now contains the updated field?

File: src/routes/accountsRouter.js [hunk 3]
```diff
@@ -13,6 +21,21 @@ apiRouter.use('/login', loginRouter);
...
+);
+apiRouter.post(
+	'/upload',
+	verifyToken,
+	fileUpload.single('image'),
+	cloudinaryConfig,
+	UploadMiddleware,
+	asyncHandler(handlePictureUpload)
+);
 apiRouter.use('/', resetPasswordRouter);
```
Comment (Yvad60) [inline src/routes/accountsRouter.js:39]: Good job!

File: src/userApp/profileUpdateValidation.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		.isLength({ min: 3 })
+		.withMessage('min_3_char'),
+	body('lastName', 'last_name_required')
+		.optional()
+		.notEmpty()
+		.withMessage('last_name_required')
+		.bail()
+		.isLength({ min: 3 })
+		.withMessage('min_3_char'),
+	body('mobileNumber', 'mobile_number_required').optional().notEmpty(),
```
Comment (Yvad60) [inline src/userApp/profileUpdateValidation.js]: Can you validate this field so that the user can't save invalid phone number

**Review by Yvad60 — DISMISSED** [review]
Approved!
nice work Mr fab with this PR implementation, features are working properly and I noticed some improvements for the existing features

**Review by samyshyaka — DISMISSED** [review]
Great work @feyton, I think this is a great implementation and the PR is ready to be merged

**Review by Yvad60 — APPROVED** [review]

**Review by samyshyaka — APPROVED** [review]

**Pull request merged**